### PR TITLE
Allow to create new enrollments when having default dates

### PR DESCRIPTION
**Pull request by mafesan**
Before these changes, when a default enrollment existed for a given unique identity and organization (e.g 01234, Example, 1900-01-01, 2100-01-01) the function enroll didn't allow to
create other enrollments within the same range with more specific dates (e.g. 2004-01-01, 2006-01-01), raising an `DuplicateRangeError`.

Now, these default start and end dates are replaced when a more specific date is set if the flag `force` is active.

Note: This PR aims to fix https://github.com/chaoss/grimoirelab-sortinghat/issues/269.

### CODE REVIEW

File: sortinghat/core/api.py [hunk 1]
```diff
@@ -607,7 +607,7 @@ def delete_domain(ctx, domain_name):
 
 
 @django.db.transaction.atomic
-def enroll(ctx, uuid, organization, from_date=None, to_date=None):
+def enroll(ctx, uuid, organization, from_date=None, to_date=None, ignore_default=False):
```
Comment (sduenas) [inline sortinghat/core/api.py:610]: I think we should rename `ignore_default` to `force` or something like that to overwrite default ranges. It's shorter and easier to remember - specially for the command line arguments.

File: sortinghat/core/errors.py [hunk 2]
```diff
@@ -90,3 +91,10 @@ class LockedIdentityError(BaseError):
 
     code = CODE_LOCKED_IDENTITY_ERROR
     message = "UniqueIdentity %(uuid)s is locked"
+
+
+class EnrollmentRangeError(BaseError):
+    """Exception raised when setting an enrollment with a wrong
+     or conflicting date range"""
+    code = CODE_ENROLLMENT_RANGE_ERROR
```
[the remark this answers is not in the document]
Comment (mafesan) [reply sortinghat/core/errors.py:99]: My bad, I thought this blank line was not needed when the docstring doesn't fit in one line.

File: sortinghat/core/errors.py [hunk 3]
```diff
@@ -90,3 +91,10 @@ class LockedIdentityError(BaseError):
 
     code = CODE_LOCKED_IDENTITY_ERROR
     message = "UniqueIdentity %(uuid)s is locked"
+
+
+class EnrollmentRangeError(BaseError):
```
[the remark this answers is not in the document]
Comment (mafesan) [reply sortinghat/core/errors.py:96]: I have renamed it to `DuplicateRangeError`, but I'm not totally convinced about this name. Other options could be `ExistingRangeError` or `OverlappedRangeError`. WDYT?

File: sortinghat/core/api.py [hunk 4]
```diff
@@ -625,6 +625,9 @@ def enroll(ctx, uuid, organization, from_date=None, to_date=None):
 
     If the given period for that enrollment is enclosed by one already
     stored, the function will raise an `EnrollmentRangeError` exception.
+    If the option `ignore_default` is set to `True`, and the existing
+    enrollment is not composed by any of the default
+    start or end date.
```
[the remark this answers is not in the document]
Comment (mafesan) [reply sortinghat/core/api.py:630]: Oops! My bad :sweat_smile:

File: sortinghat/core/api.py (hunk 1 above)
Comment (mafesan) [reply sortinghat/core/api.py:610]: Ok, I have renamed it also in the mutation.

File: sortinghat/core/errors.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (sduenas) [reply sortinghat/core/errors.py:96]: `OverlappedRangeError` sounds good to me

**Comment by mafesan** [comment]
@sduenas I have just submitted the changes after your comments. Thanks!

File: sortinghat/core/api.py [hunk 5]
```diff
@@ -639,7 +642,7 @@ def enroll(ctx, uuid, organization, from_date=None, to_date=None):
     :raises InvalidValueError: raised in three cases, when either identity or
         organization are None or empty strings; when "from_date" < 1900-01-01 or
         "to_date" > 2100-01-01; when "from_date > to_date".
-    :raises AlreadyExistsError: raised when the given period for that enrollment
+    :raises EnrollmentRangeError: raised when the given period for that enrollment
```
Comment (sduenas) [inline sortinghat/core/api.py:652]: This should be `DuplicateRangeError`.

Comment (mafesan) [reply sortinghat/core/api.py]: Oops! Fixed.

File: sortinghat/cli/cmds/enroll.py [hunk 6]
```diff
@@ -61,6 +64,10 @@ def enroll(ctx, uuid, organization, from_date, to_date, **extra):
 
     If the given period for that enrollment is enclosed by
     one already stored, the command will return an error.
+    This might happen when default values are already set
+    an enrollment. In the case, instead of removing an
```
Comment (mafesan) [inline sortinghat/cli/cmds/enroll.py:68]: I think this should be: (...)values are already set `for`/`in` an enrollment. 
WDYT?

**Pull request merged**